core/templates/domain/skill/skill-backend-api.service.ts

```diff
@@ -16,124 +16,142 @@
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skill = angular.copy(response.data.skill);
-        var groupedSkillSummaryDicts = angular.copy(
-          response.data.grouped_skill_summaries);
-        if (successCallback) {
-          successCallback({
-            skill: skill,
-            groupedSkillSummaries: groupedSkillSummaryDicts
-          });
-        }
-      }, function(errorResponse) {
+@Injectable({
+  providedIn: 'root'
+})
+export class SkillBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  private _fetchSkill(
+      skillId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: EditableSkillResponseConfig) => successCallback({
+        skill: data.skill,
+        groupedSkillSummaries: data.grouped_skill_summaries
+      }),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _fetchMultiSkills = function(skillIds, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        SKILL_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+      }
+    );
+  }
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skills = angular.copy(response.data.skills);
-        if (successCallback) {
-          successCallback(skills);
-        }
-      }, function(errorResponse) {
+  private _fetchMultiSkills(
+      skillIds: Array<string>,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.SKILL_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      }
+    );
+    this.http.get(skillDataUrl).toPromise().then(
+      (data: EditableSkillResponseConfig) => successCallback(data.skills),
+      error => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.error);
         }
-      });
-    };
-
-    var _updateSkill = function(
-        skillId, skillVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableSkillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+      }
+    );
+  }
+  private _updateSkill(
+      skillId: string, skillVersion: number, commitMessage: string,
+      changeList: Array<ChangeList>,
```

Comment:
ChangeList is an interface? If yes then follow the pattern to write interface name. Also, this name seems very generic, we use ChangeList in lots of places and all of them have a different structures, make to more specific.